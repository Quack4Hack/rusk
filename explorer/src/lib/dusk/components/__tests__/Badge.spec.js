import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { Badge } from "..";

describe("Badge", () => {
  const baseProps = {
    text: "Badge",
  };

  afterEach(cleanup);

  it('should render the Badge component using the type "neutral" as a default', () => {
    const { container } = render(Badge, baseProps);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Badge component using the type "warning" variant', () => {
    const { container } = render(Badge, { ...baseProps, variant: "warning" });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Badge component using the type "error" variant', () => {
    const { container } = render(Badge, { ...baseProps, variant: "error" });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Badge component using the type "success" variant', () => {
    const { container } = render(Badge, { ...baseProps, variant: "success" });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should pass additional class names and attributes to the root element", () => {
    const props = {
      ...baseProps,
      className: "foo bar",
      id: "some-id",
    };
    const { container } = render(Badge, { ...props });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should only display a string 10 characters long", () => {
    const props = {
      ...baseProps,
      maxlength: 10,
      text: "ABCDEFGHIJK",
    };
    const { container } = render(Badge, { ...props });
    expect(container.firstChild?.textContent?.length).toBe(10);
  });
});
