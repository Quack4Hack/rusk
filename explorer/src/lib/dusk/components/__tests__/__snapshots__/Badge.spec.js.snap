// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > should render the Badge component using the type "error" variant 1`] = `
<div>
  <span
    class="dusk-badge dusk-badge--variant-error"
  >
    Badge
  </span>
  <!--&lt;Badge&gt;-->
</div>
`;

exports[`Badge > should render the Badge component using the type "neutral" as a default 1`] = `
<div>
  <span
    class="dusk-badge dusk-badge--variant-neutral"
  >
    Badge
  </span>
  <!--&lt;Badge&gt;-->
</div>
`;

exports[`Badge > should render the Badge component using the type "success" variant 1`] = `
<div>
  <span
    class="dusk-badge dusk-badge--variant-success"
  >
    Badge
  </span>
  <!--&lt;Badge&gt;-->
</div>
`;

exports[`Badge > should render the Badge component using the type "warning" variant 1`] = `
<div>
  <span
    class="dusk-badge dusk-badge--variant-warning"
  >
    Badge
  </span>
  <!--&lt;Badge&gt;-->
</div>
`;
